webapp/src/js/app.js

```diff
@@ -117,6 +121,18 @@ _.templateSettings = {
   };
   app.constant('POUCHDB_OPTIONS', POUCHDB_OPTIONS);
 
+  const dbName = () => {
+    const xhr = new XMLHttpRequest();
+    xhr.open('GET', '/api/db', false);
+    xhr.send(null);
+    if (xhr.status === 200) {
+      const res = JSON.parse(xhr.responseText);
+      return res.name;
+    }
+  };
+
+  app.constant('DB_NAME', dbName());
```

Comment:
I think this will cause problems when running offline because you won't get the db name until you refresh the browser.

I'm still keen on a solution where the browser doesn't know the name of the DB. I think we should hardcode some string to use in the browser. At the moment I suspect all DBs are called "medic" so for backwards compatibility we should make this hardcoded db name "medic". Then in API when a request comes in for `/medic/...` we can proxy that straight on to `${environment.db}` (or whatever). That should simplify the front end a lot.